test/setup.js

```diff
@@ -91,21 +90,81 @@ function screenshotCreateOrCompare (app, t, name) {
+  if (Buffer.compare(bufActual, bufExpected) === 0) return true
+
+  // Otherwise, compare pixel by pixel
+  const pngA = PNG.sync.read(bufActual)
+  const pngE = PNG.sync.read(bufExpected)
+  if (pngA.width !== pngE.width || pngA.height !== pngE.height) return false
+  const w = pngA.width
+  const h = pngE.height
+  const da = pngA.data
+  const de = pngE.data
+  for (let y = 0; y < h; y++) {
+    for (let x = 0; x < w; x++) {
+      const i = (y * w + x) * 4
+      if (de[i + 3] === 0) continue // Skip transparent pixels
+      if (da[i] !== de[i] || da[i + 1] !== de[i + 1] || da[i + 2] !== de[i + 2]) return false
+    }
+  }
+  return true
+}
+
 // Resets the test directory, containing config.json, torrents, downloads, etc
-function wipeTestDataDir () {
-  fs.removeSync(TEST_DATA_DIR)
-  fs.mkdirpSync(TEST_DOWNLOAD_DIR) // Downloads/ is inside of TEST_DATA_DIR
+function resetTestDataDir () {
+  fs.removeSync(config.TEST_DIR)
+  // Create TEST_DIR as well as /Downloads and /Desktop
+  fs.mkdirpSync(config.TEST_DIR_DOWNLOAD)
+  fs.mkdirpSync(config.TEST_DIR_DESKTOP)
+}
+
+function deleteTestDataDir () {
+  fs.removeSync(config.TEST_DIR)
 }
 
+// Checks a given folder under Downloads.
+// Makes sure that the filenames match exactly.
+// If `filenames` is null, asserts that the folder doesn't exist.
 function compareDownloadFolder (t, dirname, filenames) {
-  const dirpath = path.join(TEST_DOWNLOAD_DIR, dirname)
+  const dirpath = path.join(config.TEST_DIR_DOWNLOAD, dirname)
   try {
     const actualFilenames = fs.readdirSync(dirpath)
     const expectedSorted = filenames.slice().sort()
     const actualSorted = actualFilenames.slice().sort()
     t.deepEqual(actualSorted, expectedSorted, 'download folder contents: ' + dirname)
   } catch (e) {
-    console.error(e)
-    t.equal(filenames, null, 'download folder missing: ' + dirname)
+    if (e.code === 'ENOENT') {
+      t.equal(filenames, null, 'download folder missing: ' + dirname)
+    } else {
+      console.error(e)
+      t.fail('unexpected error getting download folder: ' + dirname)
+    }
   }
 }
+
+// Makes sure two files have identical contents
+function compareFiles (t, pathActual, pathExpected) {
+  const bufActual = fs.readFileSync(pathActual)
+  const bufExpected = fs.readFileSync(pathExpected)
+  const match = Buffer.compare(bufActual, bufExpected) === 0
+  t.ok(match, 'correct contents: ' + pathActual)
+}
+
+// Makes sure two torrents have the same infohash and flags
+function compareTorrentFiles (t, pathActual, pathExpected) {
+  const bufActual = fs.readFileSync(pathActual)
+  const bufExpected = fs.readFileSync(pathExpected)
+  const fieldsActual = extractImportantFields(parseTorrent(bufActual))
+  const fieldsExpected = extractImportantFields(parseTorrent(bufExpected))
+  t.deepEqual(fieldsActual, fieldsExpected, 'torrent contents: ' + pathActual)
+}
+
+function extractImportantFields (parsedTorrent) {
+  const { infoHash, name, announce, urlList, comment } = parsedTorrent
```

Comment:
`name` is encapsulated within `infoHash`, but doesn't hurt to check it I suppose. The others do not affect the info hash, so good call on checking them.